composition-js/src/compose.ts

```diff
@@ -45,66 +45,105 @@ function validateCompositionOptions(options: CompositionOptions) {
+      return { errors: satisfiabilityResult.errors };
+    }
   }
 
   // printSchema calls validateOptions, which can throw
   let supergraphSdl;
   try {
     supergraphSdl = printSchema(
-      supergraph.schema,
+      mergeResult.supergraph,
       options.sdlPrintOptions ?? shallowOrderPrintedDefinitions(defaultPrintOptions),
     );
   } catch (err) {
     return { errors: [err] };
   }
 
   return {
-    schema: supergraph.schema,
+    schema: mergeResult.supergraph,
     supergraphSdl,
-    hints: mergeResult.hints.concat(hints ?? []),
+    hints: [...mergeResult.hints, ...(satisfiabilityResult?.hints ?? [])],
   };
 }
 
-export function composeServices(services: ServiceDefinition[], options: CompositionOptions = {}): CompositionResult  {
+/**
+ * Method to validate and compose services
+ * 
+ * @param services List of Service definitions
+ * @param options CompositionOptions
+ * @param runSatisfiability Flag to toggle satisfiability check within composition. Defaults to `true`
+ * @returns CompositionResult
+ */
+export function composeServices(services: ServiceDefinition[], options: CompositionOptions = {}, runSatisfiability = true): CompositionResult  {
   const subgraphs = subgraphsFromServiceList(services);
   if (Array.isArray(subgraphs)) {
     // Errors in subgraphs are not truly "composition" errors, but it's probably still the best place
     // to surface them in this case. Not that `subgraphsFromServiceList` do ensure the errors will
     // include the subgraph name in their message.
     return { errors: subgraphs };
   }
-  return compose(subgraphs, options);
+
+  return compose(subgraphs, options, runSatisfiability);
+}
+
+type SatisfiabilityArgs = {
+  supergraphSchema: Schema
+  supergraphSdl?: never
+} | { supergraphSdl: string, supergraphSchema?: never };
+
+/**
+ * Run satisfiability check for a supergraph
+ * 
+ * Can pass either the supergraph's Schema or SDL to validate
+ * @param args: SatisfiabilityArgs 
+ * @returns { errors? : GraphQLError[], hints? : CompositionHint[] }
+ */
+export function validateSatisfiability({ supergraphSchema, supergraphSdl} : SatisfiabilityArgs) : {
+  errors? : GraphQLError[],
+  hints? : CompositionHint[],
+} {
+  // We pass `null` for the `supportedFeatures` to disable the feature support validation. Validating feature support
+  // is useful when executing/handling a supergraph, but here we're just validating the supergraph we've just created,
+  // and there is no reason to error due to an unsupported feature.
+  const supergraph = supergraphSchema ? new Supergraph(supergraphSchema, null) : Supergraph.build(supergraphSdl);
+  const supergraphQueryGraph = buildSupergraphAPIQueryGraph(supergraph);
+  const federatedQueryGraph = buildFederatedQueryGraph(supergraph, false);
+  return validateGraphComposition(supergraph.schema, supergraph.subgraphNameToGraphEnumValue(), supergraphQueryGraph, federatedQueryGraph);
+}
+
+export function validateSubgraphsAndMerge(subgraphs: Subgraphs){
```

Comment:
Do you really want to export this function? I think it should be internal only, yes?

Reply:
addressed on c28b45a5